extra/modules/fiftyone-devicedetection/src/test/java/org/prebid/server/hooks/modules/fiftyone/devicedetection/v1/hooks/FiftyOneDeviceDetectionRawAuctionRequestHookTest.java

```diff
@@ -0,0 +1,1151 @@
+package org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.hooks;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.BrandVersion;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.UserAgent;
+import fiftyone.pipeline.core.flowelements.Pipeline;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.hooks.execution.v1.auction.AuctionInvocationContextImpl;
+import org.prebid.server.hooks.execution.v1.auction.AuctionRequestPayloadImpl;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.model.boundary.CollectedEvidence;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.model.config.AccountFilter;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.model.config.ModuleConfig;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.FiftyOneDeviceDetectionModule;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.core.DeviceEnricher;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.core.EnrichmentResult;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.model.ModuleContext;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.hooks.v1.auction.RawAuctionRequestHook;
+import org.prebid.server.settings.model.Account;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class FiftyOneDeviceDetectionRawAuctionRequestHookTest {
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private ModuleConfig moduleConfig;
+    private RawAuctionRequestHook target;
+    private BiFunction<Device, CollectedEvidence, EnrichmentResult> deviceRefiner;
+
+    @Before
+    public void setUp() {
+        moduleConfig = new ModuleConfig();
+        deviceRefiner = (bidRequest, evidence) -> null;
+        target = new FiftyOneDeviceDetectionRawAuctionRequestHook(
+                moduleConfig,
+                new DeviceEnricher(mock(Pipeline.class)) {
+                    @Override
+                    public EnrichmentResult populateDeviceInfo(
+                            Device device,
+                            CollectedEvidence collectedEvidence) {
+                        return deviceRefiner.apply(device, collectedEvidence);
+                    }
+                });
```

Comment:
just mock the enricher

Reply:
the cases that check the headers and so on should be in the SecureHeadersRetrieverTest, not here

here only the hook logic should be tested

Reply:
these have been addressed with the latest commit (below), thanks